pkg/plugins/backendplugin/manager.go

```diff
@@ -195,6 +196,19 @@ func (m *manager) CollectMetrics(ctx context.Context, pluginID string) (*backend
 
 // CheckHealth checks the health of a registered backend plugin.
 func (m *manager) CheckHealth(ctx context.Context, pluginContext backend.PluginContext) (*backend.CheckHealthResult, error) {
+	var dsURL string
+	if pluginContext.DataSourceInstanceSettings != nil {
+		dsURL = pluginContext.DataSourceInstanceSettings.URL
+	}
+
+	err := m.PluginRequestValidator.Validate(dsURL, nil)
+	if err != nil {
+		return &backend.CheckHealthResult{
```

Comment:
Alternatively, we may return a typed (sentinel) error instead and add the corresponding error handling on the `CheckHealth` consumer's side. Additionally, we may move the validation also to the `CheckHealth` consumer's side (the same applies for the `CallResource` method), but that would imply adding the validation twice (once per consumer: `plugins.go` & `datasources.go`).